src/main/java/org/elasticsearch/script/ScriptedOp.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.script;
+
+import java.util.Locale;
+
+/**
+ * Operation/api that uses a script as part of its execution.
+ * Note that the suggest api is considered part of search for simplicity, as well as the percolate api.
+ */
+public enum ScriptedOp {
```

Comment:
fetch falls under search, as well as percolator and suggester (this is documented). mapping covers the transfom mapping feature. The rationale is try not to have too many categories, I discussed this with @clintongormley but maybe we can change this.

The thing around alternate names is mainly that I know people are going to either use `mapping` or `mappings` and I want to be nice to them since it has the same meaning :) same for aggs and aggregations, those are synonyms as far as I'm concerned